src/Compilers/CSharp/Portable/Symbols/NullableAnnotationExtensions.cs

```diff
@@ -52,5 +53,28 @@ internal static CodeAnalysis.NullableAnnotation ToPublicAnnotation(this Nullable
             Debug.Assert((CodeAnalysis.NullableAnnotation)(NullableAnnotation.Oblivious + 1) == CodeAnalysis.NullableAnnotation.Disabled);
             return (CodeAnalysis.NullableAnnotation)annotation + 1;
         }
+
+        internal static NullabilityInfo ToNullabilityInfo(this CodeAnalysis.NullableAnnotation annotation, TypeSymbol type)
+        {
+            if (annotation == CodeAnalysis.NullableAnnotation.NotApplicable)
+            {
+                return default;
+            }
+
+            NullableAnnotation internalAnnotation = (NullableAnnotation)annotation - 1;
+            return internalAnnotation.ToNullabilityInfo(type);
+        }
+
+        internal static NullabilityInfo ToNullabilityInfo(this NullableAnnotation annotation, TypeSymbol type)
+        {
+            var flowState = TypeWithAnnotations.Create(type, annotation).ToTypeWithState().State;
+            return new NullabilityInfo(annotation.ToPublicAnnotation(), flowState.ToPublicFlowState());
+        }
+
+        internal static NullableAnnotation ToInternalAnnotation(this CodeAnalysis.NullableAnnotation annotation)
+        {
+            Debug.Assert((CodeAnalysis.NullableAnnotation)(NullableAnnotation.Oblivious + 1) == CodeAnalysis.NullableAnnotation.Disabled);
+            return annotation == CodeAnalysis.NullableAnnotation.NotApplicable ? NullableAnnotation.Oblivious : (NullableAnnotation)(annotation - 1);
```

Comment:
>CodeAnalysis.NullableAnnotation.NotApplicable ? NullableAnnotation.Oblivious : (NullableAnnotation)(annotation - 1) [](start = 33, length = 115)

Consider using a `switch` instead, to avoid depending on the actual values of the internal enum.